src/System.Windows.Forms/src/System/Windows/Forms/AxHost.State.cs

```
@@ -274,23 +271,25 @@ private void InitializeBufferFromStream(Stream ids)
             internal unsafe State? RefreshStorage(IPersistStorage.Interface iPersistStorage)
             {
                 Debug.Assert(_storage is not null, "how can we not have a storage object?");
-                Debug.Assert(_iLockBytes is not null, "how can we have a storage w/o ILockBytes?");
-                if (_storage is null || _iLockBytes is null)
+                Debug.Assert(_lockBytes is not null, "how can we have a storage w/o ILockBytes?");
+                if (_storage is null || _lockBytes is null)
                 {
                     return null;
                 }
 
-                iPersistStorage.Save(_storage, fSameAsLoad: true);
-                _storage->Commit(0);
+                using var storage = _storage.GetInterface();
+                iPersistStorage.Save(storage, fSameAsLoad: true);
+                storage.Value->Commit(0);
                 iPersistStorage.HandsOffStorage();
                 try
                 {
                     _buffer = null;
                     _memoryStream = null;
-                    _iLockBytes->Stat(out STATSTG stat, STATFLAG.STATFLAG_NONAME);
+                    using var lockBytes = _lockBytes.GetInterface();
+                    lockBytes.Value->Stat(out STATSTG stat, STATFLAG.STATFLAG_NONAME);
                     _length = (int)stat.cbSize;
                     _buffer = new byte[_length];
-                    PInvoke.GetHGlobalFromILockBytes(_iLockBytes, out nint hglobal).ThrowOnFailure();
+                    PInvoke.GetHGlobalFromILockBytes(lockBytes, out nint hglobal).ThrowOnFailure();
                     void* pointer = PInvoke.GlobalLock(hglobal);
```

A `NULL` return value is a condition that warrants a check of `GetLastError()` typically. Is there something here that makes that unnecessary?

---

Good catch. The original code was not the best at error checking and we're a bit hamstrung by having to maintain compatibility. The big issue is that we can't report errors that we didn't report before. We can, however, optimize things sometimes by short circuiting, but we have to be pretty careful to maintain the same exceptions if there are any.

In this case I'm pulling the Unlock try into the `not null` case.

For those following along, a notable exception that we _have_ been changing is throwing `ArgumentException` (of some sort) instead of `NullReferenceException`. The justification there is that while that _might_ break some, the odds are slim and the gain in diagnosability is worth it.